get-resource.sh

```diff
@@ -86,7 +86,11 @@ else
 
     if [ -n "$IP_OPTIONS" ] ; then
         BOOT_DISK=$(LIBGUESTFS_BACKEND=direct virt-filesystems -a "$RHCOS_IMAGE_FILENAME_QCOW" -l | grep boot | cut -f1 -d" ")
-        LIBGUESTFS_BACKEND=direct virt-edit -a "$RHCOS_IMAGE_FILENAME_QCOW" -m "$BOOT_DISK" /boot/loader/entries/ostree-1-rhcos.conf -e "s/^options/options ${IP_OPTIONS}/"
+        if [[ $RHCOS_IMAGE_FILENAME_QCOW =~ .*fedora* ]]; then
```

Comment:
This required qcow to always have `fedora` in the name, which is not very reliable.

Several better options:
* Use a wildcard in `virt-edit` params (not sure if it supports it)
* List files via `virt-ls` and edit every `ostree-1-.*.conf` file

Reply:
Something like this?

```bash
    if [ -n "$IP_OPTIONS" ] ; then
        BOOT_DISK=$(LIBGUESTFS_BACKEND=direct virt-filesystems -a "$RHCOS_IMAGE_FILENAME_QCOW" -l | grep boot | cut -f1 -d" ")
        BOOT_ENTRIES=$(LIBGUESTFS_BACKEND=direct virt-ls -a "$RHCOS_IMAGE_FILENAME_QCOW" "$BOOT_DISK" /boot/loader/entries)
        for BOOT_ENTRY in $BOOT_ENTRIES; do
          if [[ $BOOT_ENTRY =~ ostree-1-.*.conf ]] ; then
            LIBGUESTFS_BACKEND=direct virt-edit -a "$RHCOS_IMAGE_FILENAME_QCOW" -m "$BOOT_DISK" "/boot/loader/entries/$BOOT_ENTRY" -e "s/^options/options ${IP_OPTIONS}/"
          fi
        done
    fi
```

Reply:
Yeah, looks much cleaner. Lets add a comment about `BOOT_ENTRIES` with smth like
> \# Iterate all available ostree boot configurations to support RHCOS/FCOS seamlessly

Reply:
Ok! I'll test it and make a commit after that!

Reply:
Tested!

```
+ break
+ [[ .xz == .gz ]]
+ [[ .xz == .xz ]]
+ unxz fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2.xz
+ '[' -n ip=dhcp ']'
++ LIBGUESTFS_BACKEND=direct
++ virt-filesystems -a fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2 -l
++ grep boot
++ cut -f1 '-d '
+ BOOT_DISK=/dev/sda3
++ LIBGUESTFS_BACKEND=direct
++ virt-ls -a fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2 -m /dev/sda3 /boot/loader/entries
+ BOOT_ENTRIES=ostree-1-fedora-coreos.conf
+ for BOOT_ENTRY in $BOOT_ENTRIES
+ [[ ostree-1-fedora-coreos.conf =~ ostree-1-.*.conf ]]
+ LIBGUESTFS_BACKEND=direct
+ virt-edit -a fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2 -m /dev/sda3 /boot/loader/entries/ostree-1-fedora-coreos.conf -e 's/^options/options ip=dhcp/'
+ ln -sf fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2 cached-fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2
+ md5sum cached-fedora-coreos-35.20220327.3.0-openstack.x86_64.qcow2
+ cut -f 1 -d ' '
```